src/neptune/new/handler.py

```diff
@@ -254,42 +258,172 @@ def log(
-                elif is_string_like(first_value):
-                    warn_once(
-                        message="The object you're logging will be implicitly cast to a string."
-                        " We'll end support of this behavior in `neptune-client==1.0.0`."
-                        " To log the object as a string, use `.log(str(object))` instead.",
-                        stack_level=3,
-                    )
-                    attr = StringSeries(self._container, parse_path(self._path))
-                else:
-                    raise TypeError("Value of unsupported type {}".format(type(first_value)))
-
+                    raise ValueError("Value cannot be a collection")
+                attr = self._create_new_attribute(value)
                 self._container.set_attribute(self._path, attr)
             attr.log(value, step=step, timestamp=timestamp, wait=wait, **kwargs)
 
+    def _create_new_attribute(self, value):
+        if is_float(value):
+            return FloatSeries(self._container, parse_path(self._path))
+        elif is_dict_like(value):
+            return Namespace(self._container, parse_path(self._path))
+        elif is_string(value):
+            return StringSeries(self._container, parse_path(self._path))
+        elif FileVal.is_convertable(value):
+            return FileSeries(self._container, parse_path(self._path))
+        elif is_float_like(value):
+            return FloatSeries(self._container, parse_path(self._path))
+        elif is_string_like(value):
+            warn_once(
+                message="The object you're logging will be implicitly cast to a string."
+                " We'll end support of this behavior in `neptune-client==1.0.0`."
+                " To log the object as a string, use `.log(str(object))` instead.",
+                stack_level=3,
+            )
+            return StringSeries(self._container, parse_path(self._path))
+        else:
+            raise TypeError("Value of unsupported type {}".format(type(value)))
+
+    def extend(
+        self,
+        values: Union[Dict[str, Iterable[Any]], Iterable[Any]],
+        steps: Optional[Iterable[float]] = None,
+        timestamps: Optional[Iterable[float]] = None,
+        wait: bool = False,
+        **kwargs,
+    ) -> None:
+        """Logs a series of values by appending the provided collection of values to the end of the series.
+
+        Available for following series field types:
+            * `FloatSeries` - series of float values
+            * `StringSeries` - series of strings
+            * `FileSeries` - series of files
+
+        When you log the first value, the type of the value determines what type of field is created.
+        For more, see the field types documentation: https://docs.neptune.ai/api/field_types
+
+        Args:
+            values: Values to be added to the series field.
+            steps: Optional indeces of the entries being appended. Must be strictly increasing.
```

Comment:
Maybe it is worth to mention that steps must be a collection? The same with timestamps.